### PR TITLE
ath79: tiny: drop swconfig from single-port devices

### PR DESCRIPTION
**Pull request by Leo-PL**
In order to save some more precious space from tiny 4M devices, remove
swconfig package from devices which lack switch configuration according
to base-files.

Build tested on: all affected devices.
Run tested on: TP-Link TL-WR703N v1, TL-WR802N v2, TL-MR3040 v2.
Signed-off-by: Lech Perczak <lech.perczak@gmail.com>

### CODE REVIEW

**Comment by Leo-PL** [comment]
Had this lying around for a while, I did this while cleaning up u-boot-envtools in ath79, which I have to rebase once again before opening a PR.

**Comment by blocktrron** [comment]
Most of these boards might be single-port but still utilize the internal 5 port switch.

Removing swconfig on switchless devices (such as TL-WA901ND) should be fine however.

**Comment by adschm** [comment]
Those devices need manual package adjustments anyway after the introduction of wpad-basic-openssl. Since removing swconfig can be done easily the same way by just adding a "-swconfig" to the device packages config line already necessary anyway, I don't think we should waste time on reviewing and maintaining this for 4M flash devices at this point.

tl;dr: NAK.

**Comment by Leo-PL** [comment]
> 
> 
> Most of these boards might be single-port but still utilize the internal 5 port switch.
> 
> Removing swconfig on switchless devices (such as TL-WA901ND) should be fine however.

Those single-port devices running using internal switch do just fine without it, like for example TL-WR902AC which has swconfig disabled already, running same SoC as WR802N, so from technical standpoint this should be fine.


> 
> 
> Those devices need manual package adjustments anyway after the introduction of wpad-basic-wolfssl. Since removing swconfig can be done easily the same way by just adding a "-swconfig" to the device packages config line already necessary anyway, I don't think we should waste time on reviewing and maintaining this for 4M flash devices at this point.
> 
> tl;dr: NAK.

This is true, but the knowledge about this possibility is lost if you have to add this manually to config. Anyway, I respect your decision.

**Comment by adschm** [comment]
> > Those devices need manual package adjustments anyway after the introduction of wpad-basic-wolfssl. Since removing swconfig can be done easily the same way by just adding a "-swconfig" to the device packages config line already necessary anyway, I don't think we should waste time on reviewing and maintaining this for 4M flash devices at this point.
> > tl;dr: NAK.
> 
> This is true, but the knowledge about this possibility is lost if you have to add this manually to config. Anyway, I respect your decision.

From my perspective, 4M flash devices are in a limited support-state right now, essentially like stable branches are. So, I would fix stuff where something actually is broken and where fixing it is reasonably simple. But I would rather not do cosmetic or conceptual changes anymore, unless they are really trivial or are connected to changes for other devices.
